pkg/common/util/v1/testutil/pod.go

```diff
@@ -15,81 +15,99 @@
+	commonv1 "github.com/kubeflow/common/pkg/apis/common/v1"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/tools/cache"
-
-	tfv1 "github.com/kubeflow/training-operator/pkg/apis/tensorflow/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
-	// labels for pods and servers.
-	tfReplicaTypeLabel  = "replica-type"
-	tfReplicaIndexLabel = "replica-index"
+	DummyContainerName  = "dummy"
+	DummyContainerImage = "dummy/dummy:latest"
 )
 
-var (
-	controllerKind = tfv1.GroupVersion.WithKind(TFJobKind)
-)
+func NewBasePod(name string, job metav1.Object, refs []metav1.OwnerReference) *corev1.Pod {
 
-func NewBasePod(name string, tfJob *tfv1.TFJob) *v1.Pod {
-	return &v1.Pod{
+	return &corev1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:            name,
-			Labels:          GenLabels(tfJob.Name),
-			Namespace:       tfJob.Namespace,
-			OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(tfJob, controllerKind)},
+			Labels:          map[string]string{},
+			Namespace:       job.GetNamespace(),
+			OwnerReferences: refs,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  DummyContainerName,
+					Image: DummyContainerImage,
+				},
+			},
 		},
 	}
 }
 
-func NewPod(tfJob *tfv1.TFJob, typ string, index int) *v1.Pod {
-	pod := NewBasePod(fmt.Sprintf("%s-%d", typ, index), tfJob)
-	pod.Labels[tfReplicaTypeLabel] = typ
-	pod.Labels[tfReplicaIndexLabel] = fmt.Sprintf("%d", index)
+func NewPod(job metav1.Object, typ string, index int, refs []metav1.OwnerReference) *corev1.Pod {
+	pod := NewBasePod(fmt.Sprintf("%s-%s-%d", job.GetName(), typ, index), job, refs)
+	pod.Labels[commonv1.ReplicaTypeLabelDeprecated] = typ
+	pod.Labels[commonv1.ReplicaTypeLabel] = typ
+	pod.Labels[commonv1.ReplicaIndexLabelDeprecated] = fmt.Sprintf("%d", index)
+	pod.Labels[commonv1.ReplicaIndexLabel] = fmt.Sprintf("%d", index)
 	return pod
 }
 
-// create count pods with the given phase for the given tfJob
-func NewPodList(count int32, status v1.PodPhase, tfJob *tfv1.TFJob, typ string, start int32) []*v1.Pod {
-	pods := []*v1.Pod{}
+// NewPodList create count pods with the given phase for the given tfJob
+func NewPodList(count int32, status corev1.PodPhase, job metav1.Object, typ string, start int32, refs []metav1.OwnerReference) []*corev1.Pod {
+	pods := []*corev1.Pod{}
 	for i := int32(0); i < count; i++ {
-		newPod := NewPod(tfJob, typ, int(start+i))
-		newPod.Status = v1.PodStatus{Phase: status}
+		newPod := NewPod(job, typ, int(start+i), refs)
+		newPod.Status = corev1.PodStatus{Phase: status}
 		pods = append(pods, newPod)
 	}
 	return pods
 }
 
-func SetPodsStatuses(podIndexer cache.Indexer, tfJob *tfv1.TFJob, typ string, pendingPods, activePods, succeededPods, failedPods int32, restartCounts []int32, t *testing.T) {
+func SetPodsStatusesV2(client client.Client, job metav1.Object, typ string,
```

Comment:
if we remove `SetPodsStatuses `, why do we need `v2?`

Reply:
I could remove `v2` but how to remind users that the API of `SetPodsStatuses` has changed? Maybe keep `SetPodsStatuses`? What would you suggest?

Reply:
> I could remove v2 but how to remind users that the API of SetPodsStatuses has changed

I am not sure why we should remind users the API is changed. Is this func used by users?

Reply:
Because the `SetPodsStatuses` is exposed outside the package. I'm not sure if any other users other than kubeflow community are using this API or not. Let me know if you have source saying none is using this function.

Reply:
I have no preference on restoring `SetPodsStatuses` or removing `V2`. Any suggestions are welcome.

Reply:
If that's the case, I think we should remove `V2`.

Reply:
This has been addressed